homeassistant/components/media_player/universal.py

```diff
@@ -0,0 +1,433 @@
+        if SERVICE_TURN_OFF in self._cmds:
+            flags |= SUPPORT_TURN_OFF
+
+        if any([cmd in self._cmds for cmd in [SERVICE_VOLUME_UP,
+                                              SERVICE_VOLUME_DOWN]]):
+            flags |= SUPPORT_VOLUME_BUTTONS
+            flags &= ~SUPPORT_VOLUME_SET
+
+        if SERVICE_VOLUME_MUTE in self._cmds and \
+                ATTR_MEDIA_VOLUME_MUTED in self._attrs:
+            flags |= SUPPORT_VOLUME_MUTE
+
+        return flags
+
+    @property
+    def device_state_attributes(self):
+        """ Extra attributes a device wants to expose. """
+        active_child = self.active_child_state
+        return {ATTR_ACTIVE_CHILD: active_child.entity_id} \
+            if active_child else {}
+
+    def turn_on(self):
+        """ turn the media player on. """
+        self._override_or_child_service(SERVICE_TURN_ON)
+
+    def turn_off(self):
+        """ turn the media player off. """
+        self._override_or_child_service(SERVICE_TURN_OFF)
+
+    def mute_volume(self, is_volume_muted):
+        """ mute the volume. """
+        data = {ATTR_MEDIA_VOLUME_MUTED: is_volume_muted}
+        self._override_or_child_service(SERVICE_VOLUME_MUTE, **data)
+
+    def set_volume_level(self, volume_level):
+        """ set volume level, range 0..1. """
+        data = {ATTR_MEDIA_VOLUME_LEVEL: volume_level}
+        self._child_service(SERVICE_VOLUME_SET, **data)
+
+    def media_play(self):
+        """ Send play commmand. """
+        self._child_service(SERVICE_MEDIA_PLAY)
+
+    def media_pause(self):
+        """ Send pause command. """
+        self._child_service(SERVICE_MEDIA_PAUSE)
+
+    def media_previous_track(self):
+        """ Send previous track command. """
+        self._child_service(SERVICE_MEDIA_PREVIOUS_TRACK)
+
+    def media_next_track(self):
+        """ Send next track command. """
+        self._child_service(SERVICE_MEDIA_NEXT_TRACK)
+
+    def media_seek(self, position):
+        """ Send seek command. """
+        data = {ATTR_MEDIA_SEEK_POSITION: position}
+        self._child_service(SERVICE_MEDIA_SEEK, **data)
+
+    def play_youtube(self, media_id):
+        """ Plays a YouTube media. """
+        self._child_service(SERVICE_YOUTUBE_VIDEO, media_id=media_id)
+
+    def play_media(self, media_type, media_id):
+        """ Plays a piece of media. """
+        self._child_service(
+            SERVICE_PLAY_MEDIA, media_type=media_type, media_id=media_id)
+
+    def volume_up(self):
+        """ volume_up media player. """
+        self._override_or_child_service(SERVICE_VOLUME_UP)
+
+    def volume_down(self):
+        """ volume_down media player. """
+        self._override_or_child_service(SERVICE_VOLUME_DOWN)
+
+    def media_play_pause(self):
+        """ media_play_pause media player. """
+        self._child_service(SERVICE_MEDIA_PLAY_PAUSE)
+
+    def update_state(self, *_):
+        """ event to trigger a state update in HA """
+        self.update_ha_state()
```

Comment:
You can make this `self.update_ha_state(True)` once the `update` method caches the active child.